src/Uno.UI/DataBinding/BindingPath.cs

```diff
@@ -195,8 +195,14 @@ public string Path
 			{
 				if (!_disposed
 					&& _value != null
-					&& DependencyObjectStore.AreDifferent(value, _value.GetPrecedenceSpecificValue())
-				)
+					&& (
+						!_value.IsDependencyProperty
+
+						// Don't get the source value if we're not accessing a dependency property.
+						// WinUI does not read the property value before setting the value for a
+						// non-dependency property source.
```

Comment:
Does WinUI **read** the property if it's a DP? I'd guess it may not be reading it in both cases?

Reply:
It does not matter much if WinUI does it or not, since we're using DP.GetValue underneath and it never invokes user code. In this case, we need to read the value for style and animations.